### DISCUSSION TITLE
Why the api2 demo use  -t 1 paramater?

### DISCUSSION THREAD

**Discussion by GerrieWell** [google/libultrahdr#76] [Q&A]
Hi google developer,
Why the api2 demo use  -t 1 paramater?
I have been reading  all day today, but I still cannot figure out the role of the - t and - o parameters.
My yuv and p010 images are both taken with a phone camera. How should I configure these two parameters? I configured - t 1- o 3 according to the usage example, and the resulting gain map is like this. This image only highlights the window part on the Samsung S24.
![image](https://github.com/google/libultrahdr/assets/6158401/f1ad2874-40a2-4f3e-888c-cb6b40bdd90a)

If I have a - t 0 - o 2 configuration, then the gain map is like this, the entire image has been brightened up.:
![image](https://github.com/google/libultrahdr/assets/6158401/f0a71347-1568-4366-9839-06bec31725dd)

Which paramater should be right?

I cannot understand why the '- t 1' which means hlg EOTF function is used by default when the input is 8bityuv. Shouldn't 8-bit YUV Image  use linear functions as EOTF on most display devices?

**Comment by ram-mohan** [comment]
-t    10 bit input transfer function, optional. [0:linear, 1:hlg, 2:pq]

The '-t' option indicates the transfer characteristics (trc) of hdr intent of the input image. api-2 receives a p010 input image (hdr intent) and yuv420 8 bit image (sdr intent). The sdr intent transfer characteristics are assumed to be sRGB. The hdr intent characteristics are expected to be one of (hlg, pq). Basing on this information, both intents are converted to linear trc via InvOetf and gainmap is computed. The sdr image and gain map image is signalled via jpg format.

-o    output transfer function, optional. [0:linear, 1:hlg, 2:pq, 3:srgb]
The '-o' option is meant for decoder. This is used for signalling display transfer characteristics. Say if the camera captured content is pq trc but we are rendering it to a display with hlg support, you will pass the appropriate hlg trc during decode. After applying the gainmap in linear scale, an hdrOetf corresponding to option o is chosen.
